src/state.py

```diff
@@ -27,15 +32,64 @@ def __init__(self, reason: str):
         self.reason = reason
 
 
+class CharmConfigInvalidError(CharmStateBaseError):
+    """Exception raised when a charm configuration is found to be invalid.
+
+    Attributes:
+        msg: Explanation of the error.
+    """
+
+    def __init__(self, msg: str):
+        """Initialize a new instance of the CharmConfigInvalidError exception.
+
+        Args:
+            msg: Explanation of the error.
+        """
+        self.msg = msg
+
+
+class ProxyConfig(BaseModel):
```

Comment:
I'm wondering if we can get this code in ops

Reply:
What do you mean?

Reply:
Oh as in ops lib? that would be nice :D

Reply:
We are providing a class to wrap the proxy config in all our charms. It would be worth discussing the possibility of adding that to the ops library

Reply:
@canonical/charm-tech Hello! What do you think about such enhancements?

Reply:
Interesting. Assuming these env vars are intended for charm usage (I wasn't familiar with them), I'm not opposed to considering adding them to ops, but if you'd like to pursue that, please open an issue at https://github.com/canonical/operator/issues